airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigqueryChecker.kt

```diff
@@ -4,13 +4,47 @@
 
 package io.airbyte.integrations.destination.bigquery
 
-import io.airbyte.cdk.load.check.DestinationChecker
+import io.airbyte.cdk.load.check.DestinationCheckerSync
+import io.airbyte.cdk.load.command.DestinationCatalog
+import io.airbyte.cdk.load.state.SyncManager
+import io.airbyte.cdk.load.task.DestinationTaskLauncher
+import io.airbyte.cdk.load.write.WriteOperation
 import io.airbyte.integrations.destination.bigquery.spec.BigqueryConfiguration
-import javax.inject.Singleton
+import io.airbyte.integrations.destination.bigquery.typing_deduping.BigqueryFinalTableNameGenerator
+import io.airbyte.integrations.destination.bigquery.typing_deduping.BigqueryRawTableNameGenerator
+import io.airbyte.integrations.destination.bigquery.typing_deduping.toTableId
+import io.airbyte.integrations.destination.bigquery.util.BigqueryClientFactory
+import io.micronaut.context.annotation.Factory
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+import java.io.PipedOutputStream
 
-@Singleton
-class BigqueryChecker : DestinationChecker<BigqueryConfiguration> {
-    override fun check(config: BigqueryConfiguration) {
-        // TODO implement a real checker in the CDK; kill this class
-    }
+@Factory
+class BigqueryCheckerFactory(
+    private val catalog: DestinationCatalog,
+    @Named("checkInputStreamPipe") private val pipe: PipedOutputStream,
+    private val taskLauncher: DestinationTaskLauncher,
+    private val syncManager: SyncManager,
+) {
+    @Singleton
+    fun get() =
+        DestinationCheckerSync<BigqueryConfiguration>(
+            catalog,
+            pipe,
+            WriteOperation(taskLauncher, syncManager)
+        ) { config, stream ->
+            val bq = BigqueryClientFactory(config).make()
+            bq.getTable(
+                    BigqueryRawTableNameGenerator(config)
+                        .getTableName(stream.descriptor)
+                        .toTableId()
+                )
+                ?.delete()
+            bq.getTable(
+                    BigqueryFinalTableNameGenerator(config)
+                        .getTableName(stream.descriptor)
+                        .toTableId()
+                )
+                ?.delete()
```

Comment:
This feels a little like business logic in a bean definition. How do you feel about moving this cleaning into a separate bean that is testable?